File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-android-lib/src/main/java/no/entur/abt/android/token/TokenContext.java
package no.entur.abt.android.token;

import java.util.concurrent.locks.Lock;

public interface TokenContext<T> {

	String getId();

	Token getToken();

	void setToken(Token token);

	int getStrainNumber();

	int incrementStrainNumber();

	boolean isLatestStrain(int strainNumber);

	Lock getLock();
}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-ios-lib/AbtTokenCoreSDK/keystore/TokenContext.swift
import Foundation

public protocol TokenContext: AnyObject {
    var id: String { get }
    var token: Token? { get set }
    var tokenRenewer: TokenRenewer? { get set }
    var strainNumber: Int32 { get }

    func incrementStrainNumber() -> Int32

    func isLatestStrain(strainNumber: Int32) -> Bool
}
